### PR TITLE
Basic HTMLMediaElement seeking

### LINKED ISSUE #21998: Support for HTMLMediaElement seeking and related API properties
Spec:

- https://html.spec.whatwg.org/multipage/media.html#seeking
- https://html.spec.whatwg.org/multipage/media.html#offsets-into-the-media-resource

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #21998 

This allows media seeking only when the server supports byte-range requests.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22005)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2820,7 +2820,7 @@ dependencies = [
  "lazy_static 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "libc 0.2.42 (registry+https://github.com/rust-lang/crates.io-index)",
  "num_cpus 1.4.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "rand 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)",
+ "rand 0.3.22 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (Eijebong) [inline Cargo.lock]: Uh ?

Comment (ferjm) [reply Cargo.lock]: Hmm... no idea why this happened. I run `./mach cargo-update -p servo-media` as usual.

**Comment by ferjm** [comment]
The remaining WPTs failures are because we don't expose the `seekable` property yet.

@bors-servo try=wpt

File: components/script/dom/htmlmediaelement.rs [hunk 2]
```diff
@@ -670,13 +681,80 @@ impl HTMLMediaElement {
...
+        // FIXME(nox): Handle CORS setting from crossorigin attribute.
+        let document = document_from_node(self);
+        let destination = match self.media_type_id() {
+            HTMLMediaElementTypeId::HTMLAudioElement => Destination::Audio,
+            HTMLMediaElementTypeId::HTMLVideoElement => Destination::Video,
+        };
+        let mut headers = Headers::new();
+        headers.set(HyperRange::Bytes(vec![ByteRangeSpec::AllFrom(
+            offset.unwrap_or(0),
+        )]));
```
Comment (ceyusa) [inline components/script/dom/htmlmediaelement.rs]: does it worth to add this header if offset is zero?

Comment (ferjm) [reply components/script/dom/htmlmediaelement.rs]: For simplicity, yes :). A server not supporting byte range requests will simply ignore the header.

**Comment by ferjm** [comment]
With #22024 we can seek on media behind file urls. And seeking is obviously way faster. This is with a debug build:

seek_file

**Comment by ceyusa** [comment]
lgtm

r+

**Comment by CYBAI** [comment]
```
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"video\") must inherit property \"seeking\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 23993, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"video\") must inherit property \"currentTime\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 23994, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"video\") must inherit property \"fastSeek(double)\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 23995, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: calling fastSeek(double) on document.createElement(\"video\") with too few arguments must throw TypeError", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 23996, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"audio\") must inherit property \"seeking\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24047, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"audio\") must inherit property \"currentTime\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24048, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: document.createElement(\"audio\") must inherit property \"fastSeek(double)\" with the proper type", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24049, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: calling fastSeek(double) on document.createElement(\"audio\") with too few arguments must throw TypeError", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24050, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: attribute seeking", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24189, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: attribute currentTime", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24190, "action": "test_result", "expected": "FAIL"}
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "HTMLMediaElement interface: operation fastSeek(double)", "test": "/html/dom/interfaces.https.html?include=HTML.*", "line": 24191, "action": "test_result", "expected": "FAIL"}
```

**Comment by ferjm** [comment]
@bors-servo retry

- #22025

**Comment by ferjm** [comment]
@bors-servo retry
- #22026

**Pull request merged**